### PR TITLE
Update documentation build to use jekyll & Github pages

### PR DESCRIPTION
**Pull request by smurching**
Updates documentation to build with Jekyll so we can host it on Github pages. See https://smurching.github.io/spark-sklearn/ to see what the hosted docs will look like.

Since Github pages [doesn't support the custom Jekyll plugins](https://help.github.com/articles/adding-jekyll-plugins-to-a-github-pages-site/) used to build the docs, docs must be pre-built & pushed. 

This PR contains only the code changes needed to update the docs, while #67 also contains the prebuilt docs (static HTML + CSS + JS)

Summary of changes:
* docs/_plugins/copy_api_dirs.rb: Builds docs. Runs the existing `python/gen-docs.sh` script to generate docs and copies them into the appropriate (jekyll-specific) directory.
* docs/README.md: New doc build instructions
* python/docs/conf.py: Now reads spark-sklearn version from environment variables
* requirements-docs.txt: Requirements file for building docs.

To build the docs:
* Install Python & Ruby dependencies as specified in `docs/README.md`
* Run `jekyll build` from the `docs/` directory
* Run `jekyll serve` to locally host/inspect the docs.

### CODE REVIEW

File: docs/_plugins/copy_api_dirs.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#
```
Comment (jkbradley) [inline docs/_plugins/copy_api_dirs.rb]: You can remove this header.  We should follow up later to add Apache 2.0 headers to all files (but not ASF headers)

**Comment by jkbradley** [comment]
One more request: Can you please add files to .gitignore as needed?  It looks like there are some checked into git which should not be there in python/doc_gen/ too.

**Comment by jkbradley** [comment]
Can you add these too?  They are generated by the doc build:
```
docs/.buildinfo
	docs/.doctrees/
	docs/_site/
	docs/_sources/
	docs/_static/
	docs/genindex.html
	docs/index.html
	docs/objects.inv
	docs/py-modindex.html
	docs/search.html
	docs/searchindex.js

```

**Comment by smurching** [comment]
> Can you please add files to .gitignore as needed?

Sure, I've added python/doc_gen to the .gitignore.

The new build script (`copy_api_dirs.rb`) removes the python/doc_gen folder after building the python docs & copying them into `./docs/`, so I'll go ahead and delete the folder from the repo in my follow-up PR (#67, which will static doc files to the repo).

> Can you add these too? They are generated by the doc build: ...

(Unfortunately) the static website files in `./docs/` need to be checked into git since Github pages hosts them.

**Comment by smurching** [comment]
Actually, I think I can ignore some of the buildfiles in ./docs, updating now...

**Comment by smurching** [comment]
Ok cool, ignored the generated files in docs -- once this is merged I'll create a separate repo that contains/hosts the docs & add a link to the documentation website to the README. Thanks!

**Comment by jkbradley** [comment]
LGTM pending tests

**Comment by jkbradley** [comment]
Merging!
Thanks!

**Pull request merged**